Assets/RTLTMPro/Tests/EngageOrderingTests.cs

```diff
@@ -0,0 +1,194 @@
+﻿using System.Text;
+using NUnit.Framework;
+using RTLTMPro;
+
+using System;
+using UnityEngine;
+
+namespace RTLTMPro.Tests
+{
+    public class EngageOrderingTests
+    {
+        // Code to log the encoded output
+        /*
+            foreach (char c in result)
+            {
+                Debug.Log("\\u" + ((int)c).ToString("X4"));
+            }
+        */
+
+        // Tests need to replicate what happens in RTLTextMeshPro.GetFixedText()
+        //  If you change the code in that method, make the same changes to this function
+        protected string simulatedUpdateText(string text, bool Farsi, bool FixTags, bool PreserveNumbers)
+        {
+            var output = new FastStringBuilder(RTLSupport.DefaultBufferSize);
+            RTLSupport.FixRTL(text, output, Farsi, FixTags, PreserveNumbers);
+            output.Reverse();
+
+            return output.ToString();
+        }
+
+        [Test]
+        public void NumberOrder_MidEnd()
```

Comment:
You can write all these test a single one against a dataset using TestCaseSource and store are these input/expected tests in a single array of string tuples.

Reply:
Implemented in latest commit